presto-spi/src/main/java/com/facebook/presto/spi/type/TypeManager.java

```diff
@@ -33,6 +34,11 @@
      */
     List<Type> getTypes();
 
+    /**
+     * Gets all registered parametric types.
+     */
+    Map<String, ParametricType> getParametricTypes();
```

Comment:
The map key here (the lowercase name) is an implementation detail. We should make this return a `Collection<ParametricType>` (so we return a copy of `map.values()`).